scripts/make_stm32_module.py

```diff
@@ -64,260 +74,230 @@ def module(make_serie):
-            continue
-        if '_hal_' in os.path.splitext(f)[0]:
-            make_file.write("zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_" +
-                            os.path.splitext(f)[0][(os.path.splitext(f)[0].index('_hal_')+5):].upper() +
-                            " " + os.path.join("drivers/src", f) + ")\n")
-        if '_ll_' in os.path.splitext(f)[0]:
-            make_file.write("zephyr_library_sources_ifdef(CONFIG_USE_STM32_LL_" +
-                            os.path.splitext(f)[0][(os.path.splitext(f)[0].index('_ll_')+4):].upper() +
-                            " " + os.path.join("drivers/src", f) + ")\n")
-
-    make_file.close()
+    print("Create a new CMakeLists.txt file")
+
+    with open(CMakeLists, "w") as make_file:
+        if first_line:
+            make_file.write(first_line)
+        make_file.write("# Copyright (c) 2020 STMicroelectronics\n")
+        make_file.write("#\n")
+        make_file.write("# SPDX-License-Identifier: Apache-2.0\n")
+        make_file.write("\n")
+        make_Files = os.listdir(os.path.join("./stm32cube", make_seriexx, "drivers/src"))
+        make_Files.sort()
+
+        make_file.write("zephyr_library_sources(soc/system_" + make_seriexx + ".c)\n")
+        make_file.write("zephyr_library_sources(drivers/src/" + make_seriexx + "_hal.c)\n")
+        make_file.write(
+            "zephyr_library_sources(drivers/src/" + make_seriexx + "_hal_rcc.c)\n"
+        )
+
+        for filename in make_Files:
+            # also skipping  '_xxx_hal.c'
+            if "template" in filename:
+                continue
+            if "_hal_" in filename:
+                # retrieve IP name from filename, like adc,
+                # which is delimited by
+                #   * _hal_ on one side
+                #   * and on the other side
+                #       - either '.' (file extension)
+                #       - or '_' (in case of '_ex' ex: ..._hal_adc_ex.c_ex)
+                pattern = ".*_hal_([a-zA-Z0-9]*)[._].*"
+                make_file.write(
+                    "zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_"
+                    + re.sub(pattern, r"\1", filename).upper()
+                    + " drivers/src/"
+                    + filename
+                    + ")\n"
+                )
+            if "_ll_" in filename:
+                # retrieve IP name from filename, like adc,
+                # which is delimited by
+                #   * _ll_ on one side
+                #   * and on the other side
+                #       - either '.' (file extension)
+                #       - or '_' (in case of '_ex' ex: ..._ll_adc_ex.c_ex)
+                pattern = ".*_ll_([a-zA-Z0-9]*)[._].*"
+                make_file.write(
+                    "zephyr_library_sources_ifdef(CONFIG_USE_STM32_LL_"
+                    + re.sub(pattern, r"\1", filename).upper()
+                    + " drivers/src/"
+                    + filename
+                    + ")\n"
```

Comment:
Jinja2 may be used here (suggestion)

Reply:
Not done: I am familiar with regex, and I didn't exactly catch how jinga2 (which I am not familiar with) could help